scripts/microannotate_generator.py

```diff
@@ -37,9 +39,11 @@ def generate(self):
         )
         git_repo_path = os.path.basename(self.repo_url)
 
-        retry(
-            lambda: subprocess.run(
-                ["git", "clone", self.repo_url, git_repo_path], check=True
+        executor.submit(
```

Comment:
You are passing to `executor.submit` the result of executing the clone, this is not going to work.